pydantic_ai_slim/pydantic_ai/_agent_graph.py

```diff
@@ -504,6 +505,11 @@ async def _prepare_request(
         # Update the new message index to ensure `result.new_messages()` returns the correct messages
         ctx.deps.new_message_index -= len(original_history) - len(message_history)
 
+        # Ensure the last request has a timestamp (history processors may create new ModelRequest objects without one)
+        last_request = message_history[-1]
+        if isinstance(last_request, _messages.ModelRequest) and last_request.timestamp is None:
```

Comment:
Let's move the this check here, so we just have one place where we ensure the final item is a request:

https://github.com/pydantic/pydantic-ai/blob/d2b08ad12e103e81bd0a9dc0eb7f2a7c3472931d/pydantic_ai_slim/pydantic_ai/_agent_graph.py#L1331-L1332